src/plugins/str.rs

```diff
@@ -1,20 +1,22 @@
 use nu::{
     serve_plugin, CallInfo, Plugin, Primitive, ReturnSuccess, ReturnValue, ShellError, Signature,
-    SyntaxShape, Tagged, TaggedItem, Value,
+    SyntaxShape, Tagged, Tag, TaggedItem, Value,
 };
 
 #[derive(Debug, Eq, PartialEq)]
 enum Action {
     Downcase,
     Upcase,
     ToInteger,
+    Substring,
```

Comment:
You might want to go ahead and store the parameter for Substring in this enum case, so when you match later you can get it back out again

Reply:
You can see some examples of how to do this in the WebEvent at the top of: https://doc.rust-lang.org/rust-by-example/custom_types/enum.html